File: HelpMEApp/HelpMEWebPart/src/app/model/WantedDonation.ts
/**
 * Created by Hrvoje on 02.06.2016..
 */
import {CommentDonation} from "./CommentDonation";

export class WantedDonation{

  public Id: number;
  public DonationId: number;
  public Caption: string;
  public Description: string;
  public Category: string;
  public SubCategory: string;
  public Condition: string;
  public UserOwnerNameSurname: string;
  public UserWhoWantDonationID: number;
  public Photos: string[];
  public Comments: CommentDonation;


  constructor(Id:number, DonationId:number, Caption:string, Description:string, Category:string, SubCategory:string, Condition:string, UserOwnerNameSurname:string, UserWhoWantDonationID:number, Photos:string[], Comments:CommentDonation) {
      this.Id = Id;
      this.DonationId = DonationId;
      this.Caption = Caption;
      this.Description = Description;
      this.Category = Category;
      this.SubCategory = SubCategory;
      this.Condition = Condition;
      this.UserOwnerNameSurname = UserOwnerNameSurname;
      this.UserWhoWantDonationID = UserWhoWantDonationID;
      this.Photos = Photos;
      this.Comments = Comments;
      }
}

File: HelpMEApp/HelpMEServices/src/HelpMEServices/Models/WantedDonation.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace HelpMEServices.Models
{
    public class WantedDonation
    {
        public int Id { get; set; }
        public int DonationId { get; set; }
        public string Caption { get; set; }
        public string Description { get; set; }
        public string Category { get; set; }
        public string SubCategory { get; set; }
        public string Condition { get; set; }

        public string UserOwnerNameSurname { get; set; }

        [ForeignKey("UserWhoWantDonationID")]
        public int UserWhoWantDonationID { get; set; }
        public User UserWhoWantDonation { get; set; }

        public List<Photo> Photos { get; set; }

        public CommentDonation Comments { get; set; }
        
        
    }
}
